client/screens/SearchUsers.jsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Avatar, Box, Container, CssBaseline, Typography,
+} from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import AssignmentTurnedInIcon from '@material-ui/icons/AssignmentTurnedIn';
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+
+import axios from 'axios';
+import { Copyright } from '../components/Copyright';
+import { useAuth } from '../store/users/AuthContext';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    overflow: 'auto',
+    flexDirection: 'column',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+}));
+
+export function SearchUsers() {
+  const classes = useStyles();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [fieldEmpty, setFieldEmpty] = useState(true);
+  const [users, setUsers] = useState([]);
+  const { currentUser } = useAuth();
+
+  // get current user's token then query the api for it's users. Only works if token is authenticated.
+  useEffect(() => {
+    currentUser.getIdToken(true).then((idToken) => {
+      axios
+        .get('/api/users', {
+          headers: {
+            Authorization: `Bearer ${idToken}`,
+          },
+        })
+        .then((resp) => {
+          const susers = [];
+          resp.data.forEach((doc) => {
+            susers.push(doc);
+          });
+          setUsers(susers);
+        });
+    });
+  }, []);
+
+  // controls the response to input being entered in the input box
+  const handleInput = (event) => {
+    setSearchTerm(event.target.value);
+    if (event.target.value) {
+      setFieldEmpty(false);
+    } else {
+      setFieldEmpty(true);
+    }
+  };
+
+  // this filters the users based on what is entered in the search box
+  // if the box is empty, no users will be displayed.  This is accomplished by
+  // using fieldEmpty as a state inside of the List component
+  const searchedUsers = users.filter((user) => user.username.toLowerCase().includes(searchTerm.toLowerCase()));
```

Comment:
TODO as a followup - we should make searches API driven

Reply:
yeah instead of getting all the users and then filtering them. I quickly added that just to test if tokens were working. We can def change that later.